internal/overlord/servstate/manager.go

```diff
@@ -431,11 +439,14 @@ func (m *ServiceManager) Replan() ([]string, []string, error) {
 	needsRestart := make(map[string]bool)
 	var stop []string
 	for name, s := range m.services {
-		if config, ok := m.plan.Services[name]; ok {
-			if config.Equal(s.config) {
+		if newConfig, ok := m.plan.Services[name]; ok {
+			// TODO: be more conservative with restarts. Changes to metadata, log
+			// targets, dependencies, etc shouldn't require a restart.
+			// https://github.com/canonical/pebble/pull/165#discussion_r1142800922
+			if newConfig.Equal(s.config) {
 				continue
 			}
-			s.config = config.Copy() // update service config from plan
+			s.config = newConfig.Copy() // update service config from plan
```

Comment:
The changes to this for loop need to be reverted as they're applying a personal preference to code that is not otherwise changing or improving meaningfully. If we allow this as a practice, the next person can come in and change it to a third thing and we'll never stop.

The TODO can be dropped for now too, as it's a bit vague ("metadata" may cause restarts) and debatable (dependencies should probably cause restarts in some situations too). Please feel free to open an issue if you'd like to track specific concerns separate from the on going work that you already have coming in the next couple of PRs, though.